tests/integration/test_drop_database_replica/test.py

```diff
@@ -0,0 +1,215 @@
+    node4.query(
+        f"CREATE DATABASE db ENGINE = Replicated('{zk_path}'"
+        + r", '{shard}', '{replica}')"
+    )
+    node4.query("DETACH DATABASE db")
+    node4.query(
+        f"INSERT INTO system.zookeeper(name, path, value) VALUES ('active', '{zk_path}/replicas/s2|r2', '{node4_uuid}')",
+    )
+
+    assert "TIMEOUT_EXCEEDED" in node1.query_and_get_error(
+        "CREATE TABLE db.t22 (n int) ENGINE=Log",
+        settings={
+            "distributed_ddl_task_timeout": 5,
+            "distributed_ddl_output_mode": "none_only_active",
+        },
+    )
+    assert "TIMEOUT_EXCEEDED" in node1.query_and_get_error(
+        "CREATE TABLE db.t33 (n int) ENGINE=Log",
+        settings={
+            "distributed_ddl_task_timeout": 5,
+            "distributed_ddl_output_mode": "throw_only_active",
+        },
+    )
+    node1.query(
+        "CREATE TABLE db.t44 (n int) ENGINE=Log",
+        settings={
+            "distributed_ddl_task_timeout": 5,
+            "distributed_ddl_output_mode": "null_status_on_timeout_only_active",
+        },
+    )
+    node4.query("ATTACH DATABASE db")
+
+    node3.query("DETACH DATABASE db")
+    node4.query("SYSTEM DROP DATABASE replica 'r1' FROM SHARD 's2' FROM DATABASE db")
+    node3.query("ATTACH DATABASE db")
+    assert "Database is in readonly mode" in node3.query_and_get_error(
+        "CREATE TABLE db.t55 (n int) ENGINE=MergeTree",
+        settings={
+            "distributed_ddl_output_mode": "none",
+        },
+    )
+    node4.query("DROP DATABASE db SYNC")
+
+    node2.query("DETACH DATABASE db")
+    node1.query("SYSTEM DROP DATABASE REPLICA 's1|r2' FROM DATABASE db")
+    node2.query("ATTACH DATABASE db")
+    assert "Database is in readonly mode" in node2.query_and_get_error(
+        "CREATE TABLE db.t55 (n int) ENGINE=MergeTree",
+        settings={
+            "distributed_ddl_output_mode": "none",
+        },
+    )
+
+    node1.query("DETACH DATABASE db")
+    node4.query(f"SYSTEM DROP DATABASE REPLICA 's1|r1' FROM ZKPATH '{zk_path}'")
+    node1.query("ATTACH DATABASE db")
+    assert "Database is in readonly mode" in node1.query_and_get_error(
+        "CREATE TABLE db.t55 (n int) ENGINE=MergeTree",
+        settings={
+            "distributed_ddl_output_mode": "none",
+        },
+    )
+
+    node1.query(f"SYSTEM DROP DATABASE REPLICA 'dummy' FROM SHARD 'dummy'")
+
+    node1.query("DROP DATABASE db SYNC")
+    node2.query("DROP DATABASE db SYNC")
+    node3.query("DROP DATABASE db SYNC")
+
+    node4.query(
+        f"CREATE DATABASE db ENGINE = Replicated('{zk_path}'"
+        + r", '{shard}', '{replica}')"
+    )
+
+    node4.query(
+        "CREATE TABLE db.rmt (x INT) ENGINE=ReplicatedMergeTree ORDER BY x",
+        settings={
+            "distributed_ddl_output_mode": "none",
+        },
+    )
+    node4.query(f"SYSTEM DROP REPLICA 'dummy' FROM DATABASE db")
+    node4.query(f"SYSTEM DROP REPLICA 'dummy'")
```

Comment:
The final thing I would check is a Refreshable Materialized View; it behaves differently based on the database it is being created in, and if I remember correctly, it may refuse to be created in Atomic DB with some parameters.